compiler/resolution/resolveFunction.cpp

```diff
@@ -2043,7 +2043,7 @@ static void addLocalCopiesAndWritebacks(FnSymbol*  fn,
       if (!shouldAddInFormalTempAtCallSite(formal, fn)) {
         start->insertBefore(new CallExpr(PRIM_MOVE,
                                          tmp,
-                                         new CallExpr(astr_initCopy, formal)));
+                                         new CallExpr(astr_initCopy, formal, gTrue)));
```

Comment:
This should be true for INTENT_CONST_IN but not for INTENT_IN.

Reply:
Ah good catch. I have changed the implementation. However, there is still an issue when you pass a domain to a `const in` formal. We don't set its `definedConst` properly. I added a future for that, and will add to future work. I am not sure how difficult it is to handle that yet.